transform/error.go

```diff
@@ -0,0 +1,44 @@
+package transform
+
+import "encoding/json"
+
+const (
+	PluginInvalidInputError = "PluginInvalidInputError"
+	PluginRunError          = "PluginRunError"
+	PluginInvalidIOError    = "PluginInvalidIOError"
+)
+
+type PluginError struct {
```

Comment:
Can a plugin also pass back a PluginError? Do we want to consider classes of issues for that as well?

Reply:
no. my thought process is, the scope of work for each plugin is not going to be so big that a plugin might want error segeration. Which is why this PR puts all the plugin errors in `PluginRunError` type with a human-readable message and the actual error of the plugin.

Reply:
Probably worth jumping on a call and discussing